Multiple discord message support + .ipynb tests

luksquaresma opened the pull request:
Added support for jupyter notebooks testing and multiple messages on discord.

Addresses:

- Closes #7

mattpsvreis reviewed (dismissed):
Merge Conflicts, dá um rebase please!!

ryukinix reviewed (changes requested):
Vi que você adicionou o Jupyter como dependência dev, mas não vi nenhum notebook. Faltou fazer commit deles?

ryukinix commented on pyproject.toml:
```diff
@@ -33,6 +33,8 @@ dev = [
     "pytest-cov>=5.0.0",
     "types-requests>=2.32.0.20240712",
+    "notebook>=7.2.1",
+    "jupyter>=1.0.0",
```
É necessário declarar os dois? Geralmente isso é suficiente: https://github.com/ryukinix/egsis/blob/2c6cab31764bff73f4f6cf32fa8229d47f8b73c7/pyproject.toml#L39

luksquaresma replied:
Resolvi retirando o suporte pra JN em [84a1515](https://github.com/milharal-dev/nemli-nemlerei-bot/pull/21/commits/84a15157f6ef1e62926d3cd53462d1991649cf2d) e deixando externo via nix, acho que vai funcionar melhor. E sim, pra utilização via vscode pede ambos os pacotes.

luksquaresma reviewed (commented):
Feito

rdenadai commented on nemli/commands/utility/summarize.py:
```diff
@@ -22,7 +22,7 @@ async def summarize_command(
         min_value=20,
         max_value=1000,
-        default=settings.discord_max_messages,
+        default=200,
```
Question:
Pq removou a settings e fixou em 200?
O @ryukinix tinha comentado no discord da gente deixar como 100 isso, talvez para diminuir o contexto de conteúdo.

luksquaresma replied:
Posso unificar sem problemas, mas é uma lista da ordem de 20 elementos. Não parecia ser relevante pra performance.

Eu preferiria pra estabelecer uma variável e deixar solta só pra isso.

rdenadai commented on nemli/commands/utility/summarize.py:
```diff
@@ -66,9 +66,39 @@ async def summarize_command(
-        print(err)
+        response_list = response_to_list(summary, header=f"Resumo das últimas {message_count} mensagens")
+        for resp in response_list:
+            await interaction.followup.send(resp)
```
Question:
Ainda precisamos testar o que o @mattpsvreis comentou sobre essa chamada aqui correto?

rdenadai replied:
Testei localmente e ficou assim o resultado em um servidor meu pessoal.

![image](https://github.com/user-attachments/assets/875a5761-9808-4434-bde0-04a73003499e)

rdenadai commented on nemli/commands/utility/summarize.py:
```diff
@@ -66,9 +66,39 @@ async def summarize_command(
+    line_break = (
+        (
+            max([t for t in line_positions if t < msg_char_lim])
+            if any(map(lambda x: x < msg_char_lim, line_positions))
```
Suggestion:
Mover essa `map` para fora do `if`, assim evitamos de percorrer 2 vezes a mesma lista de `line_positions`.
Se eu entendi direito vc checa no `any` para ver se existe e depois faz o mesmo para pegar o `max`.

luksquaresma replied:
Posso unificar sem problemas, mas é uma lista da ordem de 20 elementos. Não parecia ser relevante pra performance.

Eu preferiria pra estabelecer uma variável e deixar solta só pra isso.

rdenadai replied:
Ok, esse for é o retorno do Chat gpt neh, é sempre pequeno mesmo.
Ok, pode me ignorar aqui...

rdenadai commented on nemli/config.py:
```diff
@@ -4,7 +4,6 @@
 class NemliSettings(BaseSettings):
     model_config = SettingsConfigDict(env_file=".env", env_file_encoding="utf-8", env_prefix="NEMLI_")
 
-    discord_max_messages: int = 100
```
O mesmo ponto q levantei no primeiro comment aqui ...

rdenadai reviewed (changes requested):
1 pergunta e 1 sugestão

ryukinix commented:
Eu acho que os commits ainda tão bagunçados, o rebase não tá apropriado.
Não acho que ele intencionalmente alterou pra 200, acho que é dessincronia.
Se tiver dificuldade com o rebase, podemos ajudar.

Em qua., 24 de jul. de 2024 08:44, ⟠ Rodolfo De Nadai <
***@***.***> escreveu:

> ***@***.**** requested changes on this pull request.
>
> 1 pergunta e 1 sugestão
> ------------------------------
>
> In nemli/commands/utility/summarize.py
> <https://github.com/milharal-dev/nemli-nemlerei-bot/pull/21#discussion_r1689619763>
> :
>
> > @@ -22,7 +22,7 @@ async def summarize_command(
>          required=False,
>          min_value=20,
>          max_value=1000,
> -        default=settings.discord_max_messages,
> +        default=200,
>
> Question:
> Pq removou a settings e fixou em 200?
> O @ryukinix <https://github.com/ryukinix> tinha comentado no discord da
> gente deixar como 100 isso, talvez para diminuir o contexto de conteúdo.
> ------------------------------
>
> In nemli/commands/utility/summarize.py
> <https://github.com/milharal-dev/nemli-nemlerei-bot/pull/21#discussion_r1689620721>
> :
>
> > @@ -66,9 +66,39 @@ async def summarize_command(
>
>          # Here we are getting the summary from the response and sending it to the user
>          summary = response.choices[0].message.content.strip()  # type: ignore
> -        await interaction.followup.send(f"## Resumo das últimas {message_count} mensagens:\n{summary[:1900]}")
> -    except Exception as err:
> -        print(err)
> +        response_list = response_to_list(summary, header=f"Resumo das últimas {message_count} mensagens")
> +        for resp in response_list:
> +            await interaction.followup.send(resp)
>
> Question:
> Ainda precisamos testar o que o @mattpsvreis
> <https://github.com/mattpsvreis> comentou sobre essa chamada aqui correto?
> ------------------------------
>
> In nemli/commands/utility/summarize.py
> <https://github.com/milharal-dev/nemli-nemlerei-bot/pull/21#discussion_r1689623852>
> :
>
> > +def response_to_list(summary, header, counter=0):
> +    counter += 1
> +    msg_char_lim = 1900
> +    msg_header = header + f", parte {counter}:\n"
> +    l_marker = "\n"
> +
> +    # Small responses
> +    if len(summary) < msg_char_lim:
> +        return [summary]
> +
> +    # Line breaks
> +    line_positions = [i for i, sub in enumerate(summary[: -len(l_marker)]) if sub == l_marker]
> +    line_break = (
> +        (
> +            max([t for t in line_positions if t < msg_char_lim])
> +            if any(map(lambda x: x < msg_char_lim, line_positions))
>
> Suggestion:
> Mover essa map para fora do if, assim evitamos de percorrer 2 vezes a
> mesma lista de line_positions.
> Se eu entendi direito vc checa no any para ver se existe e depois faz o
> mesmo para pegar o max.
> ------------------------------
>
> In nemli/config.py
> <https://github.com/milharal-dev/nemli-nemlerei-bot/pull/21#discussion_r1689624607>
> :
>
> > @@ -4,7 +4,6 @@
>  class NemliSettings(BaseSettings):
>      model_config = SettingsConfigDict(env_file=".env", env_file_encoding="utf-8", env_prefix="NEMLI_")
>
> -    discord_max_messages: int = 100
>
> O mesmo ponto q levantei no primeiro comment aqui ...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/milharal-dev/nemli-nemlerei-bot/pull/21#pullrequestreview-2196466642>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB2J57VH6WBY2CA4HALBG3TZN6HTDAVCNFSM6AAAAABLLRYMMOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCOJWGQ3DMNRUGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

luksquaresma committed:
Multiple message support for discord

Revert "refactor: Number of discord messages to summarize"

This reverts commit 237fdf797ecf16f97cf5df777e3d1c5a07f950c4.

Changed dependency structure with JNs support

JN support via venv removed + linting

luksquaresma reviewed (commented):
Resolvido.

ryukinix reviewed (approved):
LGTM. Depois arrumamos o teste automático disso
